### ISSUE TITLE
Last state not updated correctly

### ISSUE THREAD

**Issue by omerlh** [godaddy/kubernetes-external-secrets#466]
I had an issue with an external secret, causing last state to be -1. After deleting it I expected the metric to be 0 or 1, but it didn't - it was fixed only after rebooting the pod:

![image](https://user-images.githubusercontent.com/6730584/89977163-23e85580-dc73-11ea-9802-4e176c49b0c9.png)

**Comment by Flydiverny** [comment]
Duplicate of #399
